### DISCUSSION TITLE
[6.1] Understand how new version installed is shown

### DISCUSSION THREAD

**Discussion by llaville** [llaville/php-compat-info#318] [Show and tell]
With version **6.1.0** PHP_CompatInfo (and PHP_CompatInfoDB 3.17.0) both are removed deprecated `composer/package-versions-deprecated` composer plugin, and use now the [Runtime Composer v2](https://getcomposer.org/doc/07-runtime.md#installed-versions) utility.

In practice, that means when you get help of CompatInfo, you will get the version installed (even if in current development) and when you'll ask for long version you'll get the hash of commit corresponding to version.

See example in screenshot : 

**PHP_CompatInfo**

![compatinfo-v610](https://user-images.githubusercontent.com/364342/148198029-d20679f5-03ca-4c5b-a010-464c789c1778.png)

**PHP_CompatInfoDB**

![r3170](https://user-images.githubusercontent.com/364342/148198053-5775df86-c7d5-4811-a81c-c398bda86436.png)

**Comment by llaville** [comment]
Another example with PHP_CompatInfoDb 3.18 that have just started its development

![compatinfodb-v318x](https://user-images.githubusercontent.com/364342/148202835-6dd714c5-9af1-4bd7-97cb-8b10cf4f5dd9.png)

**Comment by remicollet** [comment]
> With version **6.1.0** PHP_CompatInfo (and PHP_CompatInfoDB 3.17.0) both are removed deprecated `composer/package-versions-deprecated` composer plugin, and use now the [Runtime Composer v2](https://getcomposer.org/doc/07-runtime.md#installed-versions) utility.

So why, during composer install, I got:

```
  - Installing composer/package-versions-deprecated (1.11.99.4): Extracting archive
composer/package-versions-deprecated contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "composer/package-versions-deprecated" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
```


P.S. probably pulled by some other libraries (doctrine/dbal and doctrine/orm)

**Reply by llaville** [reply]
Confirmed with command `composer why composer/package-versions-deprecated`

**Reply by llaville** [reply]
Not yet in effect, because composer 2.2 compatible releases 
- is available for `doctrine/dbal` (3.2.1) - see https://github.com/doctrine/dbal/commit/6cb7287d8ca650403bebcb919ef02b985bc7e40b
- not yet available for `doctrine/orm` - see https://github.com/doctrine/orm/commit/72edfbc2707c152b633744624e479d7eef04b1d0

**Reply by llaville** [reply]
Available now in `doctrine/orm` since [v2.10.5](https://github.com/doctrine/orm/releases/tag/2.10.5)
